frontend/src/MuiForms/TagSelectorBeta.tsx

```diff
@@ -0,0 +1,78 @@
+import { Autocomplete, Box, Chip, Stack, TextField, Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+
+export default function TagSelector(props: any) {
+  const { onChange, value: currentValue, label, formContext } = props
+
+  const _onChange = (_event: React.SyntheticEvent<Element, Event>, newValues: any) => {
+    onChange(newValues.map((value) => value))
+  }
+
+  const theme = useTheme()
+
+  return (
+    <>
+      {formContext.editMode && (
+        <>
+          <Typography sx={{ fontWeight: 'bold' }}>{label}</Typography>
+          <Autocomplete
+            multiple
+            isOptionEqualToValue={(option: any, value: any) => option === value}
+            value={currentValue || ''}
+            onChange={_onChange}
+            options={[]}
+            freeSolo
+            renderTags={(value: readonly string[], getTagProps) =>
+              value.map((option: string, index: number) => (
+                // eslint-disable-next-line react/jsx-key
```

Comment:
This eslint rule is trying to save us from having to refactor this in later versions of react. We've discussed this before and used the following solution in `EntityItem.tsx`:
```
            renderTags={(value, getTagProps) =>
              value.map((option, index) => (
                <Chip variant='outlined' label={option} {...getTagProps({ index })} key={option} />
              ))
            }
```